frontend/public/components/resource-list.tsx

```diff
@@ -18,13 +18,14 @@ import { DefaultPage, DefaultDetailsPage } from './default-resource';
 const allParams = props => Object.assign({}, _.get(props, 'match.params'), props);
 
 const ResourceListPage_ = connectToPlural((props: ResourceListPageProps) => {
-  const { flags, kindObj, kindsInFlight, modelRef, ns } = allParams(props);
+  const { flags, kindObj, kindsInFlight, modelRef, ns, plural } = allParams(props);
 
   if (!kindObj) {
     if (kindsInFlight) {
       return <LoadingBox />;
     }
-    return <ErrorPage404 />;
+    const missingType = isGroupVersionKind(plural) ? `"${kindForReference(plural)}" in "${apiVersionForReference(plural)}"` : `"${plural}"`;
+    return <ErrorPage404 message={`The server doesn't have a resource type ${missingType}. Try refreshing the page if it was recently added.`} />;
```

Comment:
@spadgett Added message to tell the user to try refreshing, which is standard practice on the web and should be a low-friction solution.